.github/actions/calculate-next-internal-version/action.yml

```diff
@@ -11,6 +11,16 @@ outputs:
 runs:
   using: "composite"
   steps:
+    - uses: actions/setup-python@v2
+    - id: get-current-action-branch
+      run: ${{ github.action_path }}/../shared-scripts/get-current-action-branch.sh
+      shell: bash
+    - uses: actions/checkout@v3
+      with:
+        repository: Alfresco/alfresco-build-tools
+        ref: ${{steps.get-current-action-branch.outputs.action-branch}}
+        path: build-tools
```

Comment:
This is the way I found to reference another Action inside this repository without hard coding the name of the branch/tag: 
1. `get-current-action-branch.sh` is resolving the current branch of this action. Note: `github.ref` cannot be used because it's resolving the branch of the workflow and not one of `build-tools`. 
2. the checkout action is cloning again build tools using the resolved branch

Is there a better way of doing it?

Reply:
I was struggling with that too and this looks a clever workaround, maybe you can move this steps on a separated action called something like 'setup-alfresco-build-tools' so that can be reused in any other composite action?

The unique downside I can think of that in this way we are enforcing that all actions coming from this repo must be at the same version (but maybe it's what we want to avoid complexity for using different versions for different actions within the same workflow)

Reply:
Unfortunately, I didn't find a way to make this reusable (other than using the shared script `get-current-action-branch.sh`). If I move it to a separate composite I'll need to reference it using a version (what this workaround is actually avoiding).

Note that not necessarily all the actions coming from this repo must be on the same version: this is only ensuring that a given Action `A` referencing another Action `B` in this repository will use `B` in the same version as `A`. However, an external workfow using `B` independently could reference `B` in a different version than `A`. For instance,

`- uses: Alfresco/alfresco-build-tools/.github/actions/run-pre-commit-for-helm@v3`
`- uses: Alfresco/alfresco-build-tools/.github/actions/setup-helm-docs@v2`

The `setup-helm-docs` used independently in the workfow will be in version `v2`, but the one referenced by `un-pre-commit-for-helm` will use `setup-helm-docs` in version `v3`.